immuno_tabular/verification/immuno_table_verification.Rmd

```diff
@@ -0,0 +1,1193 @@
+       colnames(mod2) <- c(colnames(mod2)[1:7], "mod2 2.5 %", "mod2 97.5 %")
+       
+       mod3 <- svyby(~bind_response_4LD, by=~VarName+Group+Visit+Trt_lb+Baseline+Marker, design, survey::svyciprop, ci=TRUE, vartype="ci", na.rm=T)
+       colnames(mod3) <- c(colnames(mod3)[1:7], "mod3 2.5 %", "mod3 97.5 %")
+       
+       res <-  rr2 %>% filter(immuno_ind==1 & !is.na(Group)) %>%
+        group_by_at(groupby_vars) %>%
+        summarise(N=n(),
+                  `Responder_1`=paste0(round(sum(response * wt.subcohort),1),"/",round(sum(wt.subcohort),1)),
+                  `% Greater than 2xLLOD_1`=paste0(round(sum(bind_response_2LD * wt.subcohort),1),"/",round(sum(wt.subcohort),1)),
+                  `% Greater than 4xLLOD_1`=paste0(round(sum(bind_response_4LD * wt.subcohort),1),"/",round(sum(wt.subcohort),1))) %>%
+         left_join(mod1, by=groupby_vars) %>%
+         left_join(mod2, by=groupby_vars) %>%
+         left_join(mod3, by=groupby_vars) %>%
+         mutate(`Responder_2` = ifelse(!is.nan(`mod1 97.5 %`), paste0(decimal_f(response*100,1),"%\n(", decimal_f(`mod1 2.5 %`*100,1),"%, ", decimal_f(`mod1 97.5 %`*100,1),"%)"), paste0(decimal_f(response*100,1),"%")),
+                `% Greater than 2xLLOD_2` = ifelse(!is.nan(`mod2 97.5 %`), paste0(decimal_f(bind_response_2LD*100,1),"%\n(", decimal_f(`mod2 2.5 %`*100,1),"%, ", decimal_f(`mod2 97.5 %`*100,1),"%)"), paste0(decimal_f(bind_response_2LD*100,1),"%")),
+                `% Greater than 4xLLOD_2` = ifelse(!is.nan(`mod3 97.5 %`), paste0(decimal_f(bind_response_4LD*100,1),"%\n(", decimal_f(`mod3 2.5 %`*100,1),"%, ", decimal_f(`mod3 97.5 %`*100,1),"%)"), paste0(decimal_f(bind_response_4LD*100,1),"%"))
+                ) %>%
+         mutate(`Responder` = paste0(`Responder_1`, " = ", `Responder_2`),
+                `% Greater than 2xLLOD` = paste0(`% Greater than 2xLLOD_1`, " = ", `% Greater than 2xLLOD_2`),
+                `% Greater than 4xLLOD` = paste0(`% Greater than 4xLLOD_1`, " = ", `% Greater than 4xLLOD_2`)) %>%
+         select(VarName, Group, Visit, Trt_lb, Baseline, Marker, N, `Responder`:`% Greater than 4xLLOD`)
+      
+      rr_table2 <- res
+      colnames(rr_table2) <- c("subgroup","Group","Visit","Arm","Baseline SARS-CoV-2","Marker","N","Responder", "% Greater than 2xLLOD", "% Greater than 4xLLOD")
+      
+      # comparison
+      rr_table2$Group=ifelse(rr_table2$subgroup=="All participants", "", rr_table2$Group)
+     
+      rr_table2_list[[ind]] <- rr_table2
+      
+      ind=ind+1
+    }
+  }
+}
+
+rr_table2_all <- do.call("rbind", rr_table2_list)
+dim(rr_table2_all)==dim(tab_bind)
+
+tab_bind[] <- lapply(tab_bind, as.character)
+rr_table2_all[] <- lapply(rr_table2_all, as.character)
+
+rr_table2_all = tab_bind[,c("subgroup","Group","Visit","Arm","Baseline SARS-CoV-2","Marker")] %>% left_join(rr_table2_all, by=c("subgroup","Group","Visit","Arm","Baseline SARS-CoV-2","Marker"))
+
+testthat::expect_equal(rr_table2_all, tab_bind)
```

Comment:
Is this to make sure the order matches?

Reply:
Yes, these are for sorting orders

Reply:
@thebioengineer Does it need to be in csv format? All the files I received from Chenchen is in csv format though...

Reply:
It looks like Chenchen bundled all the tables together for you in the RDA - make sure to comment that in your report and I think it should be okay. I would prefer csv's so when one table changes Chenchen only has to send you the one updated table, but its fine.